### PR TITLE
Fix merkle block parsing error

### PR DESCRIPTION
**Pull request by torkelrogstad**
This took an absurd amount of head-bashing to figure out, but here's a fix for a bug that's taken me several hours to struggle with.

### CODE REVIEW

File: node/src/main/scala/org/bitcoins/node/util/BitcoinSpvNodeUtil.scala [hunk 1]
```diff
@@ -38,9 +38,9 @@ trait BitcoinSpvNodeUtil extends BitcoinSLogger {
               loop(newRemainingBytes, message :: accum)
             }
           case Failure(exception) =>
-            logger.debug(
-              "Failed to parse network message, could be because tcp frame isn't aligned")
-            logger.debug(exception.getMessage)
+            logger.error(
```
Comment (torkelrogstad) [inline node/src/main/scala/org/bitcoins/node/util/BitcoinSpvNodeUtil.scala:41]: Figured it was best to up the log level, as the fact that this was on debug was the reason I didn't see it while working on the SPV node

File: core/src/main/scala/org/bitcoins/core/serializers/blockchain/RawMerkleBlockSerializer.scala [hunk 2]
```diff
@@ -18,27 +18,32 @@ sealed abstract class RawMerkleBlockSerializer
...
+    val blockHeader = RawBlockHeaderSerializer.read(headerBytes)
 
+    val (txCountBytes, afterTxCount) = afterHeader.splitAt(4)
+
+    val transactionCount = UInt32.fromBytes(txCountBytes.reverse)
+    val hashCount = CompactSizeUInt.parseCompactSizeUInt(afterTxCount)
+
+    val (_, afterHashCountBytes) = afterTxCount.splitAt(hashCount.bytes.length)
+
+    println()
```
Comment (Christewart) [inline core/src/main/scala/org/bitcoins/core/serializers/blockchain/RawMerkleBlockSerializer.scala]: nit: remove

File: node/src/main/scala/org/bitcoins/node/util/BitcoinSpvNodeUtil.scala (hunk 1 above)
Comment (Christewart) [reply node/src/main/scala/org/bitcoins/node/util/BitcoinSpvNodeUtil.scala:41]: it's fairly common that TCP frames aren't aligned, so this may be noisy

**Comment by Christewart** [comment]
So i'm ok with merginng this as unit tests do pass, but I can't seem to spot the bug. Obviously the code is pretty hairy so i'm not surprised there is something wrong with it.

**Pull request merged**